src/fges.jl

```diff
@@ -0,0 +1,528 @@
+####################################################################
+# Data Structures
+####################################################################
+
+#Collection of variables to pass along to different functions in the FGES algorithm.  
+struct ParseData{M, F<:AbstractFloat}
+    data::M #orginal data
+    normAugData::M #standardized by the mean and std of each column, appended with ones column at end
+    numFeatures::Int #number of columns
+    numObservations::Int #number of rows
+    penalty::F
+
+    function ParseData(data::M, penalty) where M 
+
+        #Determine the data type of the data inputted
+        baseType = eltype(data)
+
+        #Get the dimensions of the input data
+        numObservations, numFeatures = size(data)
+
+        #Copy the data and standardize each column
+        normAugData = copy(data)
+        normAugData .-= mean(normAugData, dims=1) #subtract mean
+        normAugData ./= std(normAugData, dims=1) #divide by standard deviation
+
+        #Augment a column of ones on the end for linear regression
+        normAugData = [normAugData ones(baseType, numObservations)]
+
+        #Ensure that the penalty is the same type as the data (e.g. Float32)
+        penalty = baseType(penalty)
+
+        return new{M, baseType}(data, normAugData, numFeatures, numObservations, penalty)
+    end
+end  
+
+#Simple structure to hold the current edge, a subset of neighbors, and a score change
+Base.@kwdef mutable struct Step{A,B}
+    edge::Edge = Edge(1,2)
+    subset::Vector{A} = A[]
+    Δscore::B = zero(B)
+end
+
+
+####################################################################
+# Base Function Overloads
+####################################################################
+
+#Print method to display the Step
+function show(io::IO, newStep::Step{A,B}) where {A,B}
+    print(io, "Edge: $(newStep.edge), Subset: $(newStep.subset), Δscore: $(newStep.Δscore)")
+end
+
+#This could be dangerous. The @memoize macro has to check if ParseData is the same argument. We only ever define one immutable ParseData, so it will never change. Any equality check should be true.
+==(a::T, b::T) where T <: ParseData = true
+
+
+####################################################################
+# Main Entry point for the Algorithm
+####################################################################
+
+"""
+    fges(data; penalty = 1.0, verbose=false)
```

Comment:
```suggestion
    fges(data; penalty = 1.0, verbose=false)
    
```
I am adding these spaces after the function signature in the doc string